specific/cmo_project/models/sale.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+from openerp import fields, models, api
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    project_number = fields.Char(
+        string='Project Code',
+        readonly=True,
+        copy=False,
+    )
+    # version = fields.Integer(
+    #     string='Version',
+    #     size=4,
+    #     default=None,
+    # )
+    project_related_id = fields.Many2one(
+        'project.project',
+        string='Project',
+    )
+
+    version = fields.Char(
+        string='Version',
+    )
+
+    @api.onchange('project_related_id')
+    def _get_project_number(self):
+        project = self.env['project.project'].browse(self.project_related_id.id)
+        print(">>>>>>>>>>>>>>>>>>>>>", self.project_related_id.id, project.project_number)
+        self.project_id = project.analytic_account_id.id
+        self.project_number = project.project_number
+
+    @api.multi
+    def write(self, vals):
+        self.ensure_one()
```

Comment:
It is **WRONG** that in write() you use **self.ensure_one()**
And this is SO, right? it is not possible to ensure one.
You should use for loop for all possible record.